File: MovieSerenity/MovieSerenity.Web/Modules/ServerTypes/MovieDB/MovieForm.ts
import { StringEditor, IntegerEditor, DateEditor, EnumEditor, LookupEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";
import { GenreDialog } from "../../Administration/Genre/GenreDialog";
import { MovieKind } from "../Modules/MovieDB.Movie.MovieKind";

export interface MovieForm {
    Title: StringEditor;
    Description: StringEditor;
    Storyline: StringEditor;
    Year: IntegerEditor;
    ReleaseDate: DateEditor;
    Runtime: IntegerEditor;
    Kind: EnumEditor;
    GenreId: LookupEditor;
}

export class MovieForm extends PrefixedContext {
    static readonly formKey = 'MovieDB.Movie';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!MovieForm.init)  {
            MovieForm.init = true;

            var w0 = StringEditor;
            var w1 = IntegerEditor;
            var w2 = DateEditor;
            var w3 = EnumEditor;
            var w4 = LookupEditor;

            initFormType(MovieForm, [
                'Title', w0,
                'Description', w0,
                'Storyline', w0,
                'Year', w1,
                'ReleaseDate', w2,
                'Runtime', w1,
                'Kind', w3,
                'GenreId', w4
            ]);
        }
    }
}

[MovieKind]; // referenced types
queueMicrotask(() => [GenreDialog]); // referenced dialogs

File: MovieSerenity/MovieSerenity.Web/Modules/MovieDB/Movie/MovieForm.cs


using MovieSerenity.Modules.MovieDB.Movie;

namespace MovieSerenity.MovieDB.Forms;

[FormScript("MovieDB.Movie")]
[BasedOnRow(typeof(MovieRow), CheckNames = true)]
public class MovieForm
{
    public string Title { get; set; }
    [TextAreaEditor(Rows = 3)]
    public string Description { get; set; }

    //[DisplayName("Cast") , MovieCastEditor , IgnoreName]
    //public List<MovieCastRow> CastList { get; set; }
    [DisplayName("Cast") , MovieDB.MovieCastEditor , IgnoreName ] 
    public List<MovieCastRow> CastList { get; set; }

    [TextAreaEditor(Rows = 8 )]

    public string Storyline { get; set; }
    public int Year { get; set; }
    public DateTime ReleaseDate { get; set; }
    public int Runtime { get; set; }
    public MovieKind Kind { get; set; }
    [Width(200)]
    public List<int> GenreList { get; set; }

}